lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -450,12 +450,9 @@ def compute_group_resource_and_labels(tag)
         # Do this here to avoid having to repeat it for each record.
         match_data = @compiled_kubernetes_tag_regexp.match(tag)
         if match_data
-          group_resource.labels['container_name'] =
-            match_data['container_name']
-          %w(namespace_name pod_name).each do |field|
-            group_common_labels["#{CONTAINER_CONSTANTS[:service]}/#{field}"] =
-              match_data[field]
-          end
+          group_resource.labels['container_name'] = match_data['container_name']
+          group_resource.labels['namespace_id'] = match_data['namespace_name']
+          group_resource.labels['pod_id'] = match_data['pod_name']
```

Comment:
Ah, good to know about the filter plugin.

For the overwriting part, I think it's the other way around though. Data extracted from `tag` is set when processing group label level, while data extracted from `kubernetes` record is set later on when processing entry level labels. But I guess the order doesn't really matter If a user does decide to specify different values in different places...